src/components/snapshot/overview/ManageWorkspacesModal.tsx

```diff
@@ -0,0 +1,193 @@
+      fontSize: theme.typography.h6.fontSize,
+      color: theme.palette.primary.light,
+    },
+    overlaySpinner: {
+      opacity: 0.9,
+      position: 'absolute' as Property.Position,
+      right: 0,
+      bottom: 0,
+      left: 0,
+      width: 'initial',
+      height: 'fit-contnet',
+      overflow: 'clip',
+      backgroundColor: theme.palette.common.white,
+      zIndex: 100,
+    },
+  });
+
+type ManageWorkspaceModalProps = {
+  classes: ClassNameMap;
+  entries: SnapshotWorkspaceEntry[];
+  modalText: string;
+  removeWorkspace: any;
+  isLoading: boolean;
+};
+
+type ManageWorkspaceModalState = {
+  open: boolean;
+};
+
+const initialState: ManageWorkspaceModalState = {
+  open: false,
+};
+
+export class ManageWorkspacesModal extends React.PureComponent<
+  ManageWorkspaceModalProps,
+  ManageWorkspaceModalState
+> {
+  constructor(props: ManageWorkspaceModalProps) {
+    super(props);
+    this.state = initialState;
+  }
+
+  handleClickOpen = () => {
+    this.setState({
+      open: true,
+    });
+  };
+
+  handleClose = () => {
+    this.setState({ open: false });
+  };
+
+  render() {
+    const { classes, modalText, entries, removeWorkspace, isLoading } = this.props;
+    const { open } = this.state;
+    const button = (
+      <IconButton
+        className={classes.iconButton}
+        aria-label={modalText}
+        onClick={this.handleClickOpen}
+        disableFocusRipple={true}
+        disableRipple={true}
+      >
+        <i className="fa-solid fa-pen-circle" />
+      </IconButton>
+    );
+
+    return (
+      <span>
+        {button}
+        <Dialog
+          fullWidth
+          maxWidth="md"
+          onClose={this.handleClose}
+          aria-labelledby="customized-dialog-title"
+          open={open}
+        >
+          <DialogTitle className={classes.dialogTitle} id="customized-dialog-title">
+            {modalText}
+            <IconButton
+              aria-label="Close"
+              className={classes.closeButton}
+              onClick={this.handleClose}
+            >
+              <CloseIcon />
+            </IconButton>
+          </DialogTitle>
+          <Typography className={classes.dialogInstructions}>
+            Removing workspace readers will remove access to data for{' '}
+            <strong>
+              <em>Project-owners, Owners, Writers, Readers</em>
+            </strong>
+            . To add workspace readers use the{' '}
+            <strong>
+              <em>Export Snapshot to Terra Workspace</em>
+            </strong>{' '}
+            button.
+          </Typography>
+          <DialogContent className={classes.dialogContent}>
+            {isLoading && (
+              <LoadingSpinner
+                delay={true}
+                delayMessage="Thank you for your patience."
+                className={classes.overlaySpinner}
+              />
+            )}
+            <ManageWorkspacesView entries={entries} removeWorkspace={removeWorkspace} />
```

Comment:
Is workspace access removal (and subsequent check that policy group members are no longer returned as readers) handled in automated tests?